npm/podController.go

```diff
@@ -449,55 +468,64 @@ func (c *podController) syncAddAndUpdatePod(newPodObj *corev1.Pod) error {
 	// There are two possiblities for npmPodObj and newPodObj
 	// #1 case The same object with the same UID and the same key (namespace + name)
 	// #2 case Different objects with different UID, but the same key (namespace + name) due to missing some events for the old object
-	// Start processing three update cases - "ip address", label", and "portlist" change by comparing last applied states against current Pod states
 
-	deleteFromIPSets := []string{}
-	addToIPSets := []string{}
-	// compare cached NPMPod's IP address against newPod's IP address
+	// Dealing with #2 pod update event, the IP addresses of cached npmPod and newPodObj are different
+	// NPM should clean up existing references of cached pod obj and its IP.
+	// then, re-add new pod obj.
 	if cachedNpmPodObj.PodIP != newPodObj.Status.PodIP {
-		metrics.SendErrorLogAndMetric(util.PodID, "[syncAddAndUpdatePod] Info: Unexpected state. Pod (Namespace:%s, Name:%s, uid:%s, has cachedPodIp:%s which is different from PodIp:%s",
-			newPodObjNs, newPodObj.Name, cachedNpmPodObj.PodUID, cachedNpmPodObj.PodIP, newPodObj.Status.PodIP)
-		// cached PodIP needs to be cleaned up from all the cached labels
-		deleteFromIPSets = util.GetIPSetListFromLabels(cachedNpmPodObj.Labels)
-
-		// Assume that the pod IP will be released when pod moves to succeeded or failed state.
-		// If the pod transitions back to an active state, then add operation will re-establish the updated pod info.
-		// new PodIP needs to be added to all newLabels
-		addToIPSets = util.GetIPSetListFromLabels(newPodObj.Labels)
-
-		log.Logf("Deleting pod %s %s from ipset %s and adding pod %s to ipset %s",
-			cachedNpmPodObj.PodUID, cachedNpmPodObj.PodIP, cachedNpmPodObj.Namespace, newPodObj.Status.PodIP, newPodObjNs,
-		)
-
-		// Delete the pod from its namespace's ipset.
-		if err = ipsMgr.DeleteFromSet(cachedNpmPodObj.Namespace, cachedNpmPodObj.PodIP, podKey); err != nil {
-			return fmt.Errorf("[syncAddAndUpdatePod] Error: failed to delete pod from namespace ipset with err: %v", err)
+		metrics.SendErrorLogAndMetric(util.PodID, "[syncAddAndUpdatePod] Info: Unexpected state. Pod (Namespace:%s, Name:%s, newUid:%s) , has cachedPodIp:%s which is different from PodIp:%s",
```

Comment:
I guess we do not need this message since it can happen.

Reply:
this line was very useful in Genesys debugging, we woudl want to see why the delete and add happened for any new folks.